src/topp/LuciphorAdapter.cpp

```diff
@@ -490,7 +490,8 @@ class LuciphorAdapter :
     }
 
     //tmp_dir
-    String temp_dir = makeAutoRemoveTempDirectory_();
+    File::TempDir dir(debug_level_ >= 2);
```

Comment:
nitpicking again: to avoid a level of indirection for the reader, you could just use
`File::TempDir temp_dir (debug_level_ >= 2);`
i.e. without `String temp_dir = dir.getPath();`
and replace occurences of `temp_dir` with `temp_dir.getPath()`